msl/equipment/resources/vaisala/ptu300.py

```diff
@@ -0,0 +1,194 @@
+        """
+        celcius = True
+        p_units = []
+        allowed_units = [        # for pressure
+            'hPa', 'psia', 'inHg', 'torr', 'bara', 'barg', 'psig', 'mbar', 'mmHg', 'kPa', 'Pa', 'mmH2O', 'inH2O'
+        ]
+        check_string_m = "Output units   : metric"
+        available_units = self.query("UNIT")
+        if not available_units == check_string_m:  # confirming device is of type PTU300
+            self.raise_exception("Check correct device connected")
+
+        for quantity, u in desired_units.items():
+            if quantity == "RH":    # only option is %RH
+                self._units["RH"] = "%RH"
+
+            elif "T" in quantity:   # options are 'C, 'F
+                if 'F' in u:        # Temperature and humidity setting is done via metric or 'non metric'
+                    celcius = False
+                if celcius:
+                    r_m = self.query("UNIT M")
+                    self._units["T"] = "'C"
+                else:
+                    r_m = self.query("UNIT N")
+                    self._units["T"] = "'F"
+
+                if not (r_m == check_string_m) == celcius:
+                    self._units["T"] = None
+                    self.check_for_errors()
+
+            elif u in allowed_units:  # assume this is a pressure quantity based on the unit
+                if p_units and u not in p_units:
+                    self.raise_exception("Only one pressure unit can be set for this barometer")
+                r_p = self.query(f"UNIT P {u}")
+                if not r_p.endswith(u):
+                    self.check_for_errors()
+                self._units[quantity] = u
+                p_units.append(u)
+
+            else:  # quantity is not pressure, temperature or humidity, so ask user to set the unit manually
+                self.raise_exception(f"{u} is not able to be set for {quantity}. Please set this unit manually.")
+
+    @property
+    def units(self) -> dict[str, str]:
+        """A dictionary of measured quantities and their associated units as set on the device by :meth:`.set_units`."""
+        return self._units
+
+    def set_format(self, format: str) -> bool:
+        """Set the format of data output to follow the pattern in the format string.  For example, in the format string
+        ``4.3 P " " 3.3 T " " 3.3 RH " " SN " " #r #n``, x.y is the number of digits and decimal places of the values;
+        P, T, RH, and SN are placeholders for pressure, temperature, relative humidity, and serial number values;
+        and " ", #r, and #n represent a string constant, carriage-return, and line feed respectively.
+        Additional allowed modifiers include ERR for error flags, U5 for unit field and (optional) length,
+        TIME for time as [hh:mm:ss], and DATE for date as [yyyy-mm-dd]. For more options, refer to the manual.
+
+        :param format: string representing desired output format
+        :return: bool to indicate successful setting of format
```

Comment:
Optional: You don't actually need to return a `bool`, since if setting the format fails an error is raised. Basically, you never return `False` so perhaps don't return anything.

Reply:
Yes the bool is obsolete now so I've removed it.